### PR TITLE
feat: use webui API in NMS

### PR DESCRIPTION
**Pull request by patriciareinoso**
# Description

This PR removes the internal NMS backend and have the client interface query the webui API directly.

Breaking change. It uses new Rock version `1.0.0`

- Remove all the files under `pages/`, which correspond to the internal NMS backend.
- Create `call*Api.tsx` files, which contain the calls to the webui API (subscribers, networks slices and device groups).
- Update URL to fetch the gnb and the upf list, which is now served by the webui.
- Update URL to fetch the swagger UI, which is now served by the webui.
- Change `next.config.js` : it contains the configuration to export the NMS into static files when building. Parameters that are no longer necessary were removed.
- Redirection from `/` to `/network-configuration` is placed on `app/page.tsx`
- Remove files not used: `DeleteSubscriberButton.tsx`, `checkNetworkSliceExists.tsx`
- Rename to Aether SD-Core
- Disable API link while waiting for https://github.com/omec-project/webconsole/pull/193

This PR depends on the following PRs:
- https://github.com/omec-project/webconsole/pull/191
- https://github.com/omec-project/webconsole/pull/194

# Checklist:

- [ ] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [ ] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added tests that validate the behaviour of the software.
- [ ] I validated that new and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: utils/getNetworkSlice.tsx
Comment (gatici) [inline utils/getNetworkSlice.tsx:1]: This method is not used anywhere, it can be removed.

File: utils/callDeviceGroupApi.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+export const apiGetAllDeviceGroups = async () => {
+  try {
+    const response = await fetch(`/config/v1/device-group/`, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    return response
```
Comment (gatici) [inline utils/callDeviceGroupApi.tsx:13]:
```suggestion
return await fetch(`/config/v1/device-group/`, {
       method: "GET",
       headers: {
         "Content-Type": "application/json",
       },

```

Comment (gatici) [reply utils/callDeviceGroupApi.tsx:13]: We are using constant response other than return. Defining constants in each method seems not necessary as output can be returned directly. This comment applies to the other methods in the other files.

File: utils/callDeviceGroupApi.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    throw new Error(`Error getting device group: Invalid name provided ${name}.`);
+  }
+  try {
+    const response = await fetch(`/config/v1/device-group/${name}`, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    return response
```
Comment (gatici) [inline utils/callDeviceGroupApi.tsx:31]:
```suggestion
return await fetch(`/config/v1/device-group/${name}`, {
       method: "GET",
       headers: {
         "Content-Type": "application/json",
       },

File: utils/callDeviceGroupApi.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    throw error;
+  }
+};
+
+export const apiPostDeviceGroup = async (name: string, deviceGroupData: any) => {
+  if (!isValidDeviceGroupName(name)){
+    throw new Error(`Error updating device group: Invalid name provided ${name}.`);
+  }
+  try {
+    const response = await fetch(`/config/v1/device-group/${name}`, {
```
Comment (gatici) [inline utils/callDeviceGroupApi.tsx:43]: We can inline the response without creating a constant.

File: utils/callDeviceGroupApi.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    throw error;
+  }
+};
+
+export const apiDeleteDeviceGroup = async (name: string) => {
+  if (!isValidDeviceGroupName(name)){
+    throw new Error(`Error deleting device group: Invalid name provided ${name}.`);
+  }
+  try {
+    const response = await fetch(`/config/v1/device-group/${name}`, {
```
Comment (gatici) [inline utils/callDeviceGroupApi.tsx:62]: We can inline the response without creating a constant

File: utils/callNetworkSliceApi.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
+function isValidNetworkSliceName(name: string): boolean {
+  return /^[a-zA-Z0-9-_]+$/.test(name);
+}
+
+export const apiGetAllNetworkSlices = async () => {
+  try {
+    const networkSlicesResponse = await fetch(`/config/v1/network-slice`, {
```
Comment (gatici) [inline utils/callNetworkSliceApi.tsx:7]: We can inline the response without creating a constant

File: utils/callNetworkSliceApi.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    throw error;
+  }
+};
+
+export const apiGetNetworkSlice = async (name: string) => {
+  if (!isValidNetworkSliceName(name)){
+    throw new Error(`Error getting network slice: Invalid name provided ${name}.`);
+  }
+  try {
+    const response = await fetch(`/config/v1/network-slice/${name}`, {
```
Comment (gatici) [inline utils/callNetworkSliceApi.tsx:25]: We can inline the response without creating a constant

File: utils/callNetworkSliceApi.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    throw error;
+  }
+};
+
+export const apiPostNetworkSlice = async (name: string, sliceData: any) => {
+  if (!isValidNetworkSliceName(name)){
+    throw new Error(`Error updating network slice: Invalid name provided ${name}.`);
+  }
+  try {
+    const response = await fetch(`/config/v1/network-slice/${name}`, {
```
Comment (gatici) [inline utils/callNetworkSliceApi.tsx:43]: We can inline the response without creating a constant.

File: utils/callNetworkSliceApi.tsx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    throw error;
+  }
+};
+
+export const apiDeleteNetworkSlice = async (name: string) => {
+  if (!isValidNetworkSliceName(name)){
+    throw new Error(`Error deleting network slice: Invalid name provided ${name}.`);
+  }
+  try {
+    const response = await fetch(`/config/v1/network-slice/${name}`, {
```
Comment (gatici) [inline utils/callNetworkSliceApi.tsx:62]: We can inline the response without creating a constant.

**Review by gatici — COMMENTED** [review]
Looks good. Minor things are commented for improvement.

**Review by ghislainbourgeois — APPROVED** [review]

**Review by gatici — APPROVED** [review]

**Pull request merged**